streamparse/dsl/thrift.py

```diff
@@ -0,0 +1,564 @@
+  void uploadNewCredentials(1: string name, 2: Credentials creds) throws (1: NotAliveException e, 2: InvalidTopologyException ite, 3: AuthorizationException aze);
+
+  // need to add functions for asking about status of storms, what nodes they're running on, looking at task logs
+
+  string beginFileUpload() throws (1: AuthorizationException aze);
+  void uploadChunk(1: string location, 2: binary chunk) throws (1: AuthorizationException aze);
+  void finishFileUpload(1: string location) throws (1: AuthorizationException aze);
+
+  string beginFileDownload(1: string file) throws (1: AuthorizationException aze);
+  //can stop downloading chunks when receive 0-length byte array back
+  binary downloadChunk(1: string id) throws (1: AuthorizationException aze);
+
+  // returns json
+  string getNimbusConf() throws (1: AuthorizationException aze);
+  // stats functions
+  ClusterSummary getClusterInfo() throws (1: AuthorizationException aze);
+  TopologyInfo getTopologyInfo(1: string id) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  TopologyInfo getTopologyInfoWithOpts(1: string id, 2: GetInfoOptions options) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  TopologyPageInfo getTopologyPageInfo(1: string id, 2: string window, 3: bool is_include_sys) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  ComponentPageInfo getComponentPageInfo(1: string topology_id, 2: string component_id, 3: string window, 4: bool is_include_sys) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  //returns json
+  string getTopologyConf(1: string id) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  /**
+   * Returns the compiled topology that contains ackers and metrics consumsers. Compare {@link #getUserTopology(String id)}.
+   */
+  StormTopology getTopology(1: string id) throws (1: NotAliveException e, 2: AuthorizationException aze);
+  /**
+   * Returns the user specified topology as submitted originally. Compare {@link #getTopology(String id)}.
+   */
+  StormTopology getUserTopology(1: string id) throws (1: NotAliveException e, 2: AuthorizationException aze);
+}
+
+struct DRPCRequest {
+  1: required string func_args;
+  2: required string request_id;
+}
+
+exception DRPCExecutionException {
+  1: required string msg;
+}
+
+service DistributedRPC {
+  string execute(1: string functionName, 2: string funcArgs) throws (1: DRPCExecutionException e, 2: AuthorizationException aze);
+}
+
+service DistributedRPCInvocations {
+  void result(1: string id, 2: string result) throws (1: AuthorizationException aze);
+  DRPCRequest fetchRequest(1: string functionName) throws (1: AuthorizationException aze);
+  void failRequest(1: string id) throws (1: AuthorizationException aze);
+}
+"""
+
+storm_thrift = thriftpy.load_fp(io.StringIO(_THRIFT_STR),
```

Comment:
Yeah, I used the `storm_thrift.ComponentObject` way to write things all over the place.  There were just a few cases where it would have been nice to save more horizontal space.